File: 066.PlusOne/clywin123/plusOne.py
class Solution:
    # @param {integer[]} digits
    # @return {integer[]}
    def plusOne(self, digits):
        carry = True
        answer = []
        for i in range(len(digits)-1,-1,-1):
            if(carry):
                digits[i] += 1
                carry = False
                if(digits[i]>=10):
                    digits[i] %=10
                    carry = True
            answer.append(digits[i])
        if(carry):
            answer.append(1)
        return answer[::-1]

File: 066.PlusOne/wyrj/PlusOne.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    var i;
    for (i = digits.length - 1; i >= 0; i--) {
        if (9 !== digits[i]) {
            digits[i] += 1;
            break;
        } else {
            digits[i] = 0;
        }
    }
    if (i < 0) {
        digits.unshift(1);
    }
    return digits
};